holidays/financial/_sifma.py

```diff
@@ -0,0 +1,148 @@
+    """
+
+    observed_label = "%s (observed)"
+    supported_categories = (HALF_DAY, PUBLIC)
+    start_year = 1950
+
+    def __init__(self, *args, **kwargs):
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        kwargs.setdefault("observed_rule", SAT_TO_PREV_FRI + SUN_TO_NEXT_MON)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        self._move_holiday(self._add_new_years_day("New Year's Day"))
+
+        # Martin Luther King Jr. Day.
+        if self._year >= 1998:
+            self._add_holiday_3rd_mon_of_jan("Martin Luther King Jr. Day")
+
+        # Washington's Birthday.
+        name = "Washington's Birthday"
+        if self._year >= 1971:
+            self._add_holiday_3rd_mon_of_feb(name)
+        else:
+            self._move_holiday(self._add_holiday_feb_22(name))
+
+        # Good Friday.
+        # Bond markets close on Good Friday, though it is not a federal holiday.
+        # Note: Some SIFMA markets may treat this as an early close instead of full closure.
+        self._add_good_friday("Good Friday")
+
+        # Memorial Day.
+        name = "Memorial Day"
+        if self._year >= 1971:
+            self._add_holiday_last_mon_of_may(name)
+        else:
+            self._move_holiday(self._add_holiday_may_30(name))
+
+        # Juneteenth National Independence Day.
+        if self._year >= 2021:
+            self._move_holiday(self._add_holiday_jun_19("Juneteenth National Independence Day"))
+
+        # Independence Day (July 4).
+        self._move_holiday(self._add_holiday_jul_4("Independence Day"))
+
+        # Labor Day (1st Monday of September).
+        self._add_holiday_1st_mon_of_sep("Labor Day")
+
+        # Columbus Day.
+        name = "Columbus Day"
+        if self._year >= 1971:
+            self._add_holiday_2nd_mon_of_oct(name)
+        else:
+            self._move_holiday(self._add_holiday_oct_12(name))
+
+        # Veterans Day.
+        name = "Veterans Day"
+        if 1971 <= self._year <= 1977:
+            self._add_holiday_4th_mon_of_oct(name)
+        else:
+            self._move_holiday(self._add_remembrance_day(name))
+
+        # Thanksgiving Day (4th Thursday of November).
+        self._add_holiday_4th_thu_of_nov("Thanksgiving Day")
+
+        # Christmas Day (December 25).
+        self._move_holiday(self._add_christmas_day("Christmas Day"))
+
+    def _populate_half_day_holidays(self):
+        begin_time_label = "Markets close at 2:00 PM ET (%s)"
+
+        # Day before New Year's Day.
+        # Apply SIFMA_EARLY_CLOSE rule to next year's January 1.
+        jan_1_next = date(self._year + 1, 1, 1)
+        early_close_nye = self._get_observed_date(jan_1_next, rule=SIFMA_EARLY_CLOSE)
+        if early_close_nye:
+            self._add_holiday(begin_time_label % "New Year's Day", early_close_nye)
+
+        # Day before Good Friday (Maundy Thursday).
```

Comment:
```suggestion
        # Day before Good Friday.
```
For style consistency.